src/api/app/controllers/webui/kiwi/images_controller.rb

```diff
@@ -87,6 +90,13 @@ def set_image
       def authorize_update
         authorize @image, :update?
       end
+
+      def cleanup_non_project_repositories!
+        return unless params[:kiwi_image][:use_project_repositories] == '1'
+
+        @image.repositories.delete_all
```

Comment:
Just as a thought which we can maybe implement later:
What about not deleting the repositories but instead marking them as ``disabled`` and when the user disables ``obsrepositories`` we add these repositories again?

Reply:
@ChrisBr, It's a good idea, I will create a new card to do this in another PR.